internal/controller/impersonatorconfig/impersonator_config.go

```diff
@@ -651,6 +706,26 @@ func (c *impersonatorConfigController) createOrUpdateService(ctx context.Context
 	return err
 }
 
+func (c *impersonatorConfigController) readExternalTLSSecret(externalTLSSecretName string) (impersonationCABundle []byte, err error) {
+	secretFromInformer, err := c.secretsInformer.Lister().Secrets(c.namespace).Get(externalTLSSecretName)
+	if err != nil {
+		c.infoLog.Info("could not find externally provided TLS secret for the impersonation proxy",
+			"secretName", externalTLSSecretName)
+		return nil, err
+	}
+
+	c.infoLog.Info("found externally provided TLS secret for the impersonation proxy",
+		"secretName", externalTLSSecretName)
+
+	err = c.loadTLSCertFromSecret(secretFromInformer)
+	if err != nil {
+		plog.Error("error loading cert from externally provided TLS secret for the impersonation proxy", err)
+		return nil, err
+	}
+
+	return secretFromInformer.Data[caCrtKey], nil
```

Comment:
Before returning the value, should we `pem.Decode` it to check if it is legal PEM data? It seems like the code does that for the other values that it finds in this Secret.

Reply:
We probably should. Not sure if this is actually base64-decoding this field either. :|

Reply:
We now check via `pem.Decode`